spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java

```diff
@@ -0,0 +1,205 @@
+import org.springframework.context.SmartLifecycle;
+
+/**
+ * An admin that delegates to an {@link AdminClient} to create topics defined
+ * in the application context.
+ *
+ * @author Gary Russell
+ * @since 1.3
+ *
+ */
+public class KafkaAdmin implements ApplicationContextAware, SmartLifecycle, DisposableBean {
+
+	private static final int DEFAULT_CLOSE_TIMEOUT = 10;
+
+	private final Log logger = LogFactory.getLog(KafkaAdmin.class);
+
+	private final AdminClient adminClient;
+
+	private boolean running;
+
+	private int phase;
+
+	private boolean autoStartup = true;
+
+	private ApplicationContext applicationContext;
+
+	private int closeTimeout = DEFAULT_CLOSE_TIMEOUT;
+
+	/**
+	 * Create an instance with the provided {@link AdminClient}.
+	 * @param adminClient the client.
+	 */
+	public KafkaAdmin(AdminClient adminClient) {
+		this.adminClient = adminClient;
+	}
+
+	/**
+	 * Create an instance with an {@link AdminClient} created from the supplied
+	 * configuration.
+	 * @param config the configuration.
+	 */
+	public KafkaAdmin(Map<String, Object> config) {
+		this.adminClient = AdminClient.create(config);
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = applicationContext;
+	}
+
+	public AdminClient getAdminClient() {
+		return this.adminClient;
+	}
+
+	public void setRunning(boolean running) {
+		this.running = running;
+	}
+
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	/**
+	 * Set the close timeout in seconds. Defaults to 10 seconds.
+	 * @param closeTimeout the timeout.
+	 */
+	public void setCloseTimeout(int closeTimeout) {
+		this.closeTimeout = closeTimeout;
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			if (this.applicationContext != null) {
+				addTopicsIfNeeded(this.applicationContext.getBeansOfType(NewTopic.class, false, false).values());
+				addAclBindingsIfNeeded(
+						this.applicationContext.getBeansOfType(AclBinding.class, false, false).values());
+			}
+			this.running = true;
+		}
+	}
+
+	@Override
+	public synchronized void stop() {
+		if (this.running) {
+			this.running = false;
+		}
+	}
+
+	@Override
+	public boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		callback.run();
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		this.adminClient.close(this.closeTimeout, TimeUnit.SECONDS);
+	}
+
+	private void addTopicsIfNeeded(Collection<NewTopic> topics) {
+		if (topics.size() > 0) {
+			Map<String, NewTopic> topicNameToTopic = new HashMap<>();
+			topics.stream().forEach(t -> topicNameToTopic.compute(t.name(), (k, v) -> v = t));
```

Comment:
It doesn't need to be for master; that's why I didn't say "cherry-pick"; I will create a 1.3.x PR after merge.